### PR TITLE
Fixing ml-client build.gradle for publishing

### PR DESCRIPTION
**Pull request by amitgalitz**
### Description

Removed shadowJar so we can publish the ml-client. 
Currently the feature branch is not publishing ml-client when running `./gradlew publishToMavenLocal`
 
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dhrubo-os** [comment]
```
org.gradle.internal.execution.WorkValidationException: A problem was found with the configuration of task ':opensearch-ml-client:compileJava' (type 'JavaCompile').
```

**Comment by ylwu-amzn** [comment]
I fine tuned some build gradle https://github.com/opensearch-project/ml-commons/commit/c456f82924d1fbf9ef4258c2b49ddb3681eecbea, can you check if that can solve your problem?

**Comment by amitgalitz** [comment]
> I fine tuned some build gradle [c456f82](https://github.com/opensearch-project/ml-commons/commit/c456f82924d1fbf9ef4258c2b49ddb3681eecbea), can you check if that can solve your problem?

Do you suggest not merging my change here? It passed the CI's locally and seems to pass mostly here, just some flaky failureson some of the runs

File: client/build.gradle [hunk 1]
```diff
@@ -14,11 +14,10 @@ plugins {
 }
 
 dependencies {
-    implementation project(path: ":${rootProject.name}-spi", configuration: 'shadow')
-    compileOnly project(':opensearch-ml-common')
+    implementation project(':opensearch-ml-common')
     compileOnly group: 'org.opensearch', name: 'opensearch', version: "${opensearch_version}"
     testImplementation group: 'junit', name: 'junit', version: '4.13.2'
-    testImplementation group: 'org.mockito', name: 'mockito-core', version: '4.4.0'
+    testImplementation group: 'org.mockito', name: 'mockito-core', version: '5.3.1'
```
Comment (ylwu-amzn) [inline client/build.gradle:20]: Just ask, why improve to 5.3.1, see any issue?

**Comment by ylwu-amzn** [comment]
> > I fine tuned some build gradle [c456f82](https://github.com/opensearch-project/ml-commons/commit/c456f82924d1fbf9ef4258c2b49ddb3681eecbea), can you check if that can solve your problem?
> 
> Do you suggest not merging my change here? It passed the CI's locally and seems to pass mostly here, just some flaky failureson some of the runs

Feel free to merge, rebase before merge?

File: client/build.gradle (hunk 1 above)
Comment (amitgalitz) [reply client/build.gradle]: This was carried over from main actually where we have mockito set to 5.3.1. Its not a must a though so I reversed this for now

**Comment by dhrubo-os** [comment]
@amitgalitz do we still need this PR?

**Comment by amitgalitz** [comment]
> @amitgalitz do we still need this PR?

Okay to close this one. Yaliang's PR that merged just before also fixed the issue

**Pull request closed without merging**